machineconfiguration/v1/types.go

```diff
@@ -0,0 +1,747 @@
+package v1
+
+import (
+	configv1 "github.com/openshift/api/config/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// MachineConfigRoleLabelKey is metadata key in the MachineConfig. Specifies the node role that config should be applied to.
+// For example: `master` or `worker`
+const MachineConfigRoleLabelKey = "machineconfiguration.openshift.io/role"
+
+// KubeletConfigRoleLabelPrefix is the label that must be present in the KubeletConfig CR
+const KubeletConfigRoleLabelPrefix = "pools.operator.machineconfiguration.openshift.io/"
```

Comment:
nit: Perhaps at some point in the future, other custom resources besides KubeletConfigs would use this annotation to associate themselves with pools? Do we want a more generic name for the variable, like `PoolLabelPrefix`?

Reply:
Does this need to be fixed pre-merge?

Reply:
I vote no, it's already been in for a year: https://github.com/openshift/machine-config-operator/pull/3015 and was put in specifically for consumption by the kublet config controller

Reply:
okay, I say that, but I went and looked and...nothing references it right now. I wonder if it ever did?

I still think we should handle this later...but...yeah. Cleanup. :smile: